scipy/interpolate/_bsplines.py

```diff
@@ -561,6 +601,71 @@ def integrate(self, a, b, extrapolate=None):
         integral *= sign
         return integral.reshape(ca.shape[1:])
 
+    @classmethod
+    def from_cubic(cls, cb, axis=0, bc_type='natural', extrapolate=None):
+        """
+        Construct a polynomial in the B-spline basis
+        from a piecewise polynomial in the power basis.
+
+        Parameters
+        ----------
+        cb : CubicSpline
+            A piecewise polynomial in the power basis, as created by CubicSpline
+        axis : int, optional
+            Axis along which y is assumed to be varying. Meaning that for x[i] the
+            corresponding values are np.take(y, i, axis=axis). Default is 0.
+        bc_type : string, optional
+            Boundary condition type as in CubicSpline
+        extrapolate: {bool, ‘periodic’, None}, optional
+            If bool, determines whether to extrapolate to out-of-bounds points based
+            on first and last intervals, or to return NaNs. If ‘periodic’, periodic
+            extrapolation is used. If None (default), extrapolate is set to ‘periodic’
+            for bc_type='periodic' and to True otherwise.
+
+        Returns
+        -------
+        b : BSpline object
+            A new instance representing the initial polynomial in the B-spline basis.
+
+        Notes
+        -----
+        The algorithm follows from differentiation the Marsden’s identity [1].
+        bc_type == 'not-a-knot' is not implemented yet.
+
+        References
+        ----------
+        .. [1] Tom Lyche and Knut Mørken, Spline Methods, 2005, Section 3.1.2
+
+        """
+        x = cb.x
+        coef = cb.c
+        k = cb.c.shape[0] - 1
+        n = x.shape[0]
+        t = np.zeros(n + 2 * k)
+
+        if bc_type == 'natural' or bc_type == 'clamped':
+            t[k: -k] = x
+            for i in range(1, k + 1):
+                t[k - i] = x[0]
+                t[n + k + i - 1] = x[-1]
+        elif bc_type == 'periodic':
+            t = _periodic_knots(x, k)
+
+        c = np.zeros(n + 2, )
+        for i in range(n - 2):
+            mu = _find_left(t, x[i], k)
+            for m in range(k + 1):
+                c[mu - k] += coef[m, i] * np.power(-1, k - m) * _diff_dual_poly(mu - k, k, x[i], m, t) * \
```

Comment:
If this is in cython, beware of the += operator for `double complex[:]` --- will have to write it out `c[...] = c[...] + ...`.
I suspect this code may perfectly well stay dealing with numpy arrays without much typing, but if you do use typed memoryviews and complex numbers, this is something to keep at the back of your mind.